ambari-server/src/main/resources/common-services/ZOOKEEPER/3.4.5/package/templates/zookeeper_client_jaas.conf.j2

```diff
@@ -19,5 +19,6 @@
 Client {
 com.sun.security.auth.module.Krb5LoginModule required
 useKeyTab=false
-useTicketCache=true;
+useTicketCache=true
```

Comment:
I am not sure if this is the correct solution here.  I will help to investigate.

Reply:
at least the service check passes. There are also stack changes at `hdp_ambari_definitions`, but I did not publish the patch yet

Reply:
@dlysnichenko After playing with ZK service check for a while, I found that the solution is to add the following to the JAVA command that executes the ZK client:

```
 -Dzookeeper.sasl.client.username=<principal name>
```

I think the easiest approach is to add this to the `CLIENT_JVMFLAGS` env in `zookeeper-env/content`.  

The `<principal name>' value is the first component of the ZK principal.  If the ZK principal is "my_zk/c7401.ambari.apache.org@EXAMPLE.COM", then the value that needs to be set for principal name is "my_zk".  Once you do this, you can view the KDC log and see entries like (in the case of an MIT KDC) during the service check. 

```
Aug 30 13:23:53 c7401.ambari.apache.org krb5kdc[14444](info): TGS_REQ (4 etypes {18 17 16 23}) 192.168.74.101: ISSUE: authtime 1535635426, etypes {rep=18 tkt=23 ses=18}, ambari-qa-c1@EXAMPLE.COM for my_zk/c7401.ambari.apache.org@EXAMPLE.COM
```

My guess is that this will help any process that sources /etc/zookeeper/conf/zookeeper-env.sh before executing the ZK client.

Reply:
@rlevas ,
Thank you for hint, will update the patch